components/solr-access-service/api/src/test/java/org/phenotips/solr/HPOScriptServiceTest.java

```diff
@@ -0,0 +1,250 @@
+        expectedResult.add("HP:0001507");
+        expectedResult.add("HP:0000118");
+        expectedResult.add("HP:0000001");
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void clearClearsServerAndCache() throws ComponentLookupException, IOException, SolrServerException
+    {
+        int returnVal = this.mocker.getComponentUnderTest().clear();
+        verify(this.server).deleteByQuery("*:*");
+        verify(this.server).commit();
+        verify(this.cache).removeAll();
+        Assert.assertEquals(0, returnVal);
+    }
+
+    private void indexMockSolrDocListFromResource() throws IOException, SolrServerException, ComponentLookupException
+    {
+        CapturingMatcher<Collection<SolrInputDocument>> allTermsCap = new CapturingMatcher<>();
+        when(this.server.add(Matchers.argThat(allTermsCap))).thenReturn(new UpdateResponse());
+
+        indexReturn = this.mocker.getComponentUnderTest().index(this.getClass().getResource("/hpo-test.obo").toString(), fieldNames);
+
+        Collection<SolrInputDocument> allTerms = allTermsCap.getLastValue();
+        solrDocList = new SolrDocumentList();
+        for(SolrInputDocument i : allTerms){
+            solrDocList.add(ClientUtils.toSolrDocument(i));
+        }
+    }
+
+
+    private class QueryAnswer implements Answer<QueryResponse>{
+
+        @Override
+        public QueryResponse answer(InvocationOnMock invocationOnMock) throws Throwable
+        {
+            SolrParams params = (SolrParams) invocationOnMock.getArguments()[0];
+            if (params == null) {
+                when(response.getResults()).thenReturn(mock(SolrDocumentList.class));
+                return response;
+            }
+            for (SolrDocument item : solrDocList) {
+                String fieldValue = (String)item.getFieldValue("id");
+                fieldValue = fieldValue.replace(":", "\\:");
+                if (params.get(CommonParams.Q).contains(fieldValue)) {
+                    SolrDocumentList matchedItem = new SolrDocumentList();
+                    matchedItem.add(item);
+                    when(response.getResults()).thenReturn(matchedItem);
+                    return response;
+                }
+            }
+            when(response.getResults()).thenReturn(mock(SolrDocumentList.class));
+            return  response;
+        }
+    }
+
+    private class IsMatchingIDQuery extends ArgumentMatcher<SolrParams>{
+
+        private String id;
+
+        public IsMatchingIDQuery(String id){
+            id = id.replace(":", "\\:");
+            this.id = id;
+        }
+
+        @Override
+        public boolean matches(Object argument)
+        {
+            if (argument == null) {
+                return false;
+            }
+            SolrParams params = (SolrParams)argument;
+            return params.get(CommonParams.Q).startsWith("id")
+                    && params.get(CommonParams.Q).contains(id);
```

Comment:
You should cache `params.get(CommonParams.Q)` as a variable, so you dont have to fetch it twice. Makes it a bit easier to read.